tijanabekic/assignment3/Assignment 3,new.java

```diff
@@ -0,0 +1,167 @@
+//simple dictionary for test purpose, not the follow up solution
+public class Dictionary {
+
+    private static LinkedList<String> words;
+    private static LinkedList<String> prefixes;
+
+    public Dictionary() {
+        prefixes = new LinkedList<>();
+        prefixes.add("CARD");
+        prefixes.add("CART");
+        prefixes.add("CAR");
+        prefixes.add("CAT");
+        prefixes.add("CA");
+        prefixes.add("C");
+
+        words = new LinkedList<>();
+        words.add("CARD");
+        words.add("CART");
+        words.add("CAR");
+        words.add("CAT");
+    }
+
+    public static boolean isPrefix(String word) {
+        for (String s : prefixes) {
+            if (word.equals(s)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public static boolean isWord(String word) {
+
+        for (String s : words) {
+            if (word.equals(s)) {
+                return true;
+            }
+        }
+        return false;
+    }
+    
+    public void addWord(String word) {
+        words.add(word);
+        
+    }
+
+}
+
+
+public class A3Q1 {
+    public static Dictionary dictionary;
+    public static void checkWord(char[][] grid, int i, int j, String start, int m, int n, HashMap<String, Boolean> map, HashSet<String> set) {

```

Comment:
Maybe use names like width, height, prefix and words instead of m,n,start and set.
map and set are not very helpful names for the Map or Set as they don't explain what is inside themselves.